src/odemis/acq/test/stream_test.py

```diff
@@ -2822,7 +2822,7 @@ def on_im(im):
 
         e.clear()
         ars.background.value = calib
-        numpy.testing.assert_equal(ars.background.value, calib[0, 0, 0])
+        numpy.testing.assert_equal(ars.background.value[0], calib[0, 0, 0])
```

Comment:
This change highlights the issue of automatically converting a single DataArray with a list. It's quite confusing for the (code) user/reader. You write something and next time you read it, it's inside a list?!

I think this is a good reason to reconsider the "trick" to make the background data always a list. I'd suggest that it accepts both a None, DataArray, or list of DataArray, and keeps the value as set. The trick is only to slightly simplify the code, but if that makes the rest of the code more complicated, that's not useful.

Reply:
I changed it now so None, DataArray, or list of DataArray but I think it is not the most elegant solution. Also I am not sure what you mean with using the value as a set and where...Would be happy to have some more explanation on that :)